### PR TITLE
Added string and dispatch queue extensions

### PR DESCRIPTION
**Pull request by VatoKo**
I've taken into account all the suggestions from my previous PR and made some changes.

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.0.
- [x] New extensions support iOS 10.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [ ] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

**Comment by guykogus** [comment]
Please separate the additions into separate PRs, one for the `String` addition and another for `DispatchQueue`.

File: Sources/SwifterSwift/SwiftStdlib/StringExtensions.swift [hunk 1]
```diff
@@ -957,6 +957,17 @@ public extension String {
         return range(of: pattern, options: .regularExpression, range: nil, locale: nil) != nil
     }
     #endif
+    
+    #if canImport(Foundation)
```
Comment (LucianoPAlmeida) [inline Sources/SwifterSwift/SwiftStdlib/StringExtensions.swift:961]: String changes are still here...

File: Sources/SwifterSwift/Dispatch/DispatchQueueExtensions.swift [hunk 2]
```diff
@@ -41,6 +41,19 @@ public extension DispatchQueue {
...
         return DispatchQueue.getSpecific(key: key) != nil
     }
+    
+    /// SwifterSwift: Runs passed closure asynchronous after certain time interval
+    ///
+    /// - Parameter delay: The time inverval after which the closure will run.
+    /// - Parameter qos: Quality of service at which the work item should be executed.
+    /// - Parameter flags: Flags that control the execution environment of the work item.
+    /// - Parameter work: The closure to run after certain time interval.
+    func asyncAfter(delay: Double,
```
Comment (LucianoPAlmeida) [inline Sources/SwifterSwift/Dispatch/DispatchQueueExtensions.swift:51]: I think we can use `TimeInterval`...

File: Sources/SwifterSwift/Dispatch/DispatchQueueExtensions.swift [hunk 3]
```diff
@@ -41,6 +41,19 @@ public extension DispatchQueue {
...
+                    qos: DispatchQoS = .unspecified,
+                    flags: DispatchWorkItemFlags = [],
+                    execute work: @escaping @convention(block) () -> Void) {
```
Comment (LucianoPAlmeida) [inline Sources/SwifterSwift/Dispatch/DispatchQueueExtensions.swift:54]: I think we don't need to explicitly require a block convention here, we can have a swift closure here. As far as I know, swift to block are implicit convertible...

File: Sources/SwifterSwift/SwiftStdlib/StringExtensions.swift (hunk 1 above)
Comment (VatoKo) [reply Sources/SwifterSwift/SwiftStdlib/StringExtensions.swift:961]: As @guykogus suggested I separated these 2 extensions and made 2 separate PRs. I think this PR could be closed as those other two exist.

**Pull request closed without merging**